cmd/zpool/zpool_vdev.c

```diff
@@ -1585,8 +1592,11 @@ construct_spec(nvlist_t *props, int argc, char **argv)
 			 * construct the appropriate nvlist describing the vdev.
 			 */
 			if ((nv = make_leaf_vdev(props, argv[0],
-			    is_log)) == NULL)
+			    is_log)) == NULL) {
+				free(spares);
+				free(top);
```

Comment:
These are in the opposite order as the other free()s. I know the computer doesn't care, but for code clarity, I would suggest consistency.